metaflow/datatools/s3.py

```diff
@@ -161,6 +190,39 @@ def size(self):
         """
         return self._size
 
+    @property
+    def has_info(self):
```

Comment:
do we need this? Isn't it sufficient for `content_type` and `range_info` to return `None` if the info is not available

Reply:
Yes, could remove. I was trying to provide an easy way to know if the info was loaded (as opposed to being loaded but empty). It does the check you mention; I wanted to avoid users doing things like `if self.metadata` which can't really distinguish between (a) there is metadata but it wasn't loaded and (b) metadata was loaded and there is nothing to see.

Reply:
could `.metadata` return `{}` for loaded-but-empty and `None` for not loaded?

Reply:
Absolutely. It actually does that (I hope anyways). My point was just that both {} and None evaluate to False so the if expression is a bit of a trap but happy to remove it.

Reply:
No further comment here so leaving as is and resolving. Re-open if you would like to change.